### PR TITLE
Remove swap parameter and related code

### PR DESCRIPTION
**Pull request by stweil**
Remove all code for auto detection of endianness used in data files.
This prepares the switch to data files with fixed endianness.

Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by stweil** [comment]
@theraysmith, this PR is ready to get pulled as soon as there is a decision to switch to fixed endianness (little endian) for the training data files.

File: ccstruct/matrix.h [hunk 1]
```diff
@@ -475,26 +463,17 @@ class GENERIC_2D_ARRAY {
...
-  bool DeSerializeSize(bool swap, FILE* fp) {
+  bool DeSerializeSize(FILE* fp) {
     inT32 size1, size2;
     if (fread(&size1, sizeof(size1), 1, fp) != 1) return false;
     if (fread(&size2, sizeof(size2), 1, fp) != 1) return false;
-    if (swap) {
-      ReverseN(&size1, sizeof(size1));
-      ReverseN(&size2, sizeof(size2));
-    }
     Resize(size1, size2, empty_);
```
Comment (theraysmith) [inline ccstruct/matrix.h]: There is a malloc failure and a crash right here if you don't reverse the size to be allocated!

**Comment by theraysmith** [comment]
Rejected. I don't see how this can possibly work.
See comment on matrix.h.

**Comment by stweil** [comment]
I'm afraid there is a misunderstanding. This patch does not implement big endian support. It removes the current effort to support big endian as a first step for a different implementation which I think is more promising.

**Pull request closed without merging**